### DISCUSSION TITLE
کانفیگ های بیشتر در فایل xray

### DISCUSSION THREAD

**Discussion by M03ED** [Gozargah/Marzban#146] [Show and tell]
سلام.
با توجه به اینکه دوستان در گروه تلگرام هر روز راجع به اینکه چطور باید به فایل xray_config.json پروتکل های جدید اضافه کنند سوال میپرسند و اکثرا هم به مشکل می خورند بنده چند تا پروتکل که استفاده بیشتری داره رو داخل فایل نوشتم و براتون همینجا قرار میدم

نکته 1 : با توجه به محدود بودن پورت های نسخه رایگان کلود فلر امکان استفاده همزمان همه این پروتکل ها نیست و شما محبورید یا تعدادی رو بدون TLS روی پورت های http اجرا کنید و یا اونها رو حذف کنید.
نکته 2 : برای به مشکل نخوردن فایل تعدادی از پروتکل ها هم اکنون روی پورت های http هستند ، در صورت نیاز به TLS فراموش نکنید اون ها رو تغییر بدید.

پروتکل های موجود در فایل عبارتند از : 
Vmess : ( WS - TCP - gRPC )
Vless : ( WS - TCP - gRPC )
Trojan : ( WS - TCP - gRPC )
Shadowsocks

به امید آزادی.
[xray_config.zip](https://github.com/Gozargah/Marzban/files/11093652/xray_config.zip)

**Comment by M03ED** [comment]
Vless tcp Reality
[xray_config vless Reality.zip](https://github.com/Gozargah/Marzban/files/11424719/xray_config.vless.Reality.zip)

**Comment by rm-killer** [comment]
داادش میشه یک آموزش بدی چجوری میشه از طریق nano این تغیرات رو  انجام داد؟
من ما .env آشنایی ندارم. سرچ کردم توی نت مثالی پیدا نکردم که کار منو راه بنداره.
چجوری میشه این موارد رو طوری ادیت کرد که در نسخه UI قسمت داشبورد هم دیده بشه که شما اضافه کردید؟؟؟ ممنون میشم راهنمایی کنید

**Comment by rm-killer** [comment]
ضمنا فکر نکنم تنظیمات کانفیگ شما دیگه کار کنه؟ درسته؟ چون شروع فایل خیلی با نسخه دیفالت تنظیمات فرق داره.... نسخه شما فکر کنم برای ورژن های قدیمی هست؟؟

**Reply by M03ED** [reply]
این تنظیمات و یا همون دیفالت مرزبان ربطی به نسخه مرزبان ندارن ، این برای هسته ایکس ری هست و کار هم می‌کنه ، چیزی تغییر نکرده ، دوما که از طریق nano به هیچ وجه انجام ندید ، اگر دسترسی دارید با vs code و اگر ندارید با همون ادیتور تو پنل انجام بدید ، نکته آخر ، این تنظیمات برای فایل env نیست

**Reply by rm-killer** [reply]
درسته. پس باید فایل رو مستقیم تغییر بدم. یک سوال دیگه. به فرض من چنتا کانفیگ دیگه به این فایل اضافه کردم. ولی توی پنل بازم همون 4 تا رو نشون میده. این رو باید چیکار کنم؟؟؟ راهی نداره اونجا هم اضافه بشه؟

**Comment by ie1ya3x** [comment]
سلام میشه کانفیگ هارو بدون tls هم بزارید.

**Comment by peymansh68** [comment]
عالی
